server/src/main/java/io/crate/metadata/table/Operation.java

```diff
@@ -57,25 +61,35 @@ public enum Operation {
     public static final EnumSet<Operation> SYS_READ_ONLY = EnumSet.of(READ);
     public static final EnumSet<Operation> READ_ONLY = EnumSet.of(READ, ALTER_BLOCKS);
     public static final EnumSet<Operation> CLOSED_OPERATIONS = EnumSet.of(ALTER_OPEN, ALTER_CLOSE, ALTER_TABLE_RENAME);
-    public static final EnumSet<Operation> BLOB_OPERATIONS = EnumSet.of(READ, OPTIMIZE, ALTER, ALTER_REROUTE, DROP);
-    public static final EnumSet<Operation> READ_DISABLED_OPERATIONS = EnumSet.of(UPDATE, INSERT, DELETE, DROP, ALTER,
-        ALTER_OPEN, ALTER_CLOSE, ALTER_REROUTE, ALTER_BLOCKS, REFRESH, OPTIMIZE);
-    public static final EnumSet<Operation> WRITE_DISABLED_OPERATIONS = EnumSet.of(READ, ALTER, ALTER_OPEN, ALTER_CLOSE,
-        ALTER_BLOCKS, ALTER_REROUTE, SHOW_CREATE, REFRESH, OPTIMIZE, COPY_TO, CREATE_SNAPSHOT);
+    public static final EnumSet<Operation> BLOB_OPERATIONS = EnumSet.of(
+        READ, OPTIMIZE, ALTER, ALTER_SET_RESET, ALTER_REROUTE, DROP);
+    public static final EnumSet<Operation> READ_DISABLED_OPERATIONS = EnumSet.of(
+        UPDATE, INSERT, DELETE, DROP, ALTER, ALTER_OPEN, ALTER_CLOSE, ALTER_SET_RESET, ALTER_REROUTE, ALTER_BLOCKS,
+        REFRESH, OPTIMIZE);
+    public static final EnumSet<Operation> WRITE_DISABLED_OPERATIONS = EnumSet.of(READ, ALTER, ALTER_SET_RESET,
+        ALTER_OPEN, ALTER_CLOSE, ALTER_BLOCKS, ALTER_REROUTE, SHOW_CREATE, REFRESH, OPTIMIZE, COPY_TO, CREATE_SNAPSHOT);
     public static final EnumSet<Operation> METADATA_DISABLED_OPERATIONS = EnumSet.of(READ, UPDATE, INSERT, DELETE,
         ALTER_BLOCKS, ALTER_OPEN, ALTER_CLOSE, ALTER_REROUTE, REFRESH, SHOW_CREATE, OPTIMIZE);
     public static final EnumSet<Operation> SUBSCRIBED_IN_LOGICAL_REPLICATION = EnumSet.of(
-        READ, ALTER_BLOCKS, ALTER_REROUTE, OPTIMIZE, REFRESH, COPY_TO, SHOW_CREATE);
+        READ, ALTER_BLOCKS, ALTER_REROUTE, ALTER_SET_RESET, OPTIMIZE, REFRESH, COPY_TO, SHOW_CREATE);
     public static final EnumSet<Operation> PUBLISHED_IN_LOGICAL_REPLICATION = EnumSet.of(
-        READ, UPDATE, INSERT, DELETE, DROP, ALTER, ALTER_BLOCKS, ALTER_CLOSE, ALTER_REROUTE, REFRESH,
-        SHOW_CREATE, COPY_TO, OPTIMIZE, RESTORE_SNAPSHOT, CREATE_SNAPSHOT);
+        READ, UPDATE, INSERT, DELETE, DROP, ALTER, ALTER_SET_RESET, ALTER_BLOCKS, ALTER_CLOSE, ALTER_REROUTE,
+        REFRESH, SHOW_CREATE, COPY_TO, OPTIMIZE, RESTORE_SNAPSHOT, CREATE_SNAPSHOT);
 
     private final String representation;
 
     Operation(String representation) {
         this.representation = representation;
     }
 
+    private static final Set<String> ALLOWED_SETTINGS_TO_ALTER = Set.of(SETTING_AUTO_EXPAND_REPLICAS,
+                                                                        SETTING_NUMBER_OF_REPLICAS,
+                                                                        SETTING_INDEX_REFRESH_INTERVAL);
+
+    public static boolean isAllowedSettingsToAlter(Settings settings) {
```

Comment:
Wouldn't this block other settings on normal tables as well?

What about the suggestion of adding a new `Setting.Property`, doesn't it work? This would avoid the declaration of a dedicated list which we'd also have to keep in mind and maintain later on.

Reply:
For setting not included in the list, we fallback to `Operation.ALTER`. However, i give the `Setting.Property` a try.